packages/custom-types/player-log.d.ts

```diff
@@ -20,6 +24,43 @@ export interface PlayerLogUI {
   time?: string
 }
 
+/** a player has had some interaction with the UI */
+export interface PlainInteraction {
+  aType: string
+}
+
+/** a player has marked a message as read */
+export interface MessageReadInteraction extends PlainInteraction {
+  aType: typeof MESSAGE_READ_INTERACTION
+  _id: string
+}
+
+/** a player has marked a message as unread */
+export interface MessageUnReadInteraction extends PlainInteraction {
+  aType: typeof MESSAGE_UNREAD_INTERACTION
+  _id: string
+}
+
+/** a player has read a map annotation */
+export interface MapAnnotationReadInteraction extends PlainInteraction {
+  aType: typeof MAP_ANNOTATION_READ_INTERACTION
+  _id: string
+}
+
+/** union of all types of activity */
+export type PlayerInteraction = PlainInteraction | MessageReadInteraction | MessageUnReadInteraction | MapAnnotationReadInteraction
+
+export interface PlayerLogEntry {
+  activityTime: string
+  activityType: PlayerInteraction
```

Comment:
@lilit - instead of `activityType` as a string, we now have it as an object that can take its own metadata (such as the id of the message that has been read).